package/MDAnalysis/analysis/rms.py

```diff
@@ -535,29 +540,90 @@ def __init__(self, atomgroup, reference=None, select='all',
                         igroup, sel['reference'], sel['mobile'],
                         len(atoms['reference']), len(atoms['mobile'])))
 
-        # Explicitly check for "mass" because this option CAN
-        # be used with groupselection. (get_weights() returns the mass array
-        # for "mass")
-        if not iterable(self.weights) and self.weights == "mass":
-            pass
-        else:
-            self.weights = get_weights(self.mobile_atoms, self.weights)
 
-        # cannot use arbitrary weight array (for superposition) with
-        # groupselections because arrays will not match
-        if (len(self.groupselections) > 0 and (
-                iterable(self.weights) or self.weights not in ("mass", None))):
-            raise ValueError("groupselections can only be combined with "
-                             "weights=None or weights='mass', not a weight "
-                             "array.")
+        # check weights type
+        if (np.array(self.weights).ndim == 1) & (np.array(self.weights).dtype 
+						in (np.dtype('float64'),np.dtype('int64'))):
+            self.weights = get_weights(self.mobile_atoms, self.weights)
+        elif not iterable(self.weights):
+            self.weights = get_weights(self.mobile_atoms, self.weights)
```

Comment:
Per above, the desired behaviour for `weights='mass'` is to use `'mass'` for both `select` and `groupselections`, which is what [this](https://github.com/MDAnalysis/mdanalysis/pull/2577/files#diff-d19ae35d57b874dea08d858a08da1041R580) in `_prepare()`  is for - but if we do have `weights='mass'` it'll be overwritten here when `get_weights` returns the atomic masses and won't be picked up below.